code/modules/surgery/surgery.dm

```diff
@@ -1,22 +1,133 @@
+///Datum Surgery Helpers//
+/datum/surgery
+	var/name
+	var/status = 1
+	var/list/steps = list()
+	/*
+	var/eyes	=	0
+	var/face	=	0
+	var/appendix =	0
+	var/ribcage =	0
+	var/head_reattach = 0									//Steps in a surgery
+	*/
+
+	var/can_cancel = 1
+	var/step_in_progress = 0
+	var/list/in_progress = list()									//Actively performing a Surgery
+	var/location = "chest"										//Surgery location
+	var/requires_organic_bodypart = 1							//Prevents you from performing an operation on robotic limbs
+	var/list/possible_locs = list() 							//Multiple locations -- c0
+	var/obj/item/organ/organ_ref									//Operable body part
+	var/current_organ = "organ"
+	var/list/allowed_mob = list(/mob/living/carbon/human)
+
+/datum/surgery/proc/can_start(mob/user, mob/living/carbon/target)
+	// if 0 surgery wont show up in list
+	// put special restrictions here
+	return 1
+
+
+/datum/surgery/proc/next_step(mob/user, mob/living/carbon/target)
+	if(step_in_progress)	return
+
+	var/datum/surgery_step/S = get_surgery_step()
+	if(S)
+		if(S.try_op(user, target, user.zone_sel.selecting, user.get_active_hand(), src))
+			return 1
+	return 0
+
+/datum/surgery/proc/get_surgery_step()
+	var/step_type = steps[status]
+	return new step_type
+
+
+/datum/surgery/proc/complete(mob/living/carbon/human/target)
+	target.surgeries -= src
+	src = null
+
+
+
 /* SURGERY STEPS */
 /datum/surgery_step
 	var/priority = 0	//steps with higher priority would be attempted first
 
 	// type path referencing tools that can be used for this step, and how well are they suited for it
 	var/list/allowed_tools = null
-	// type paths referencing mutantraces that this step applies to.
-	var/list/allowed_species = null
-	var/list/disallowed_species = null
+	var/implement_type = null
 
 	// duration of the step
-	var/min_duration = 0
 	var/max_duration = 0
```

Comment:
@Aurorablade this is a deceptive var name whyyyy.

It's ok =p I'm nitpicking; don't worry about correcting it.